File: 01-week-1--starter-algorithms/02-day-3--fibonacci-series/javascript/fibonacci.js
function fibonacci(num) {
  // type your code here
  if (num <2){
    return num;
  }

  let numbers = [0,1];

  for (let i = 0; i <num - 1; ++i){
    const sum = numbers[0] + numbers[1];
    numbers = [numbers[1], sum];
  }

  return numbers[1];
  

  }


if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 0");
  console.log("=>", fibonacci(0));

  console.log("");

  console.log("Expecting: 1");
  console.log("=>", fibonacci(2));

  console.log("");

  console.log("Expecting: 55");
  console.log("=>", fibonacci(10));
}

module.exports = fibonacci;

// Pseudocode:
// create a zero index array with starting values, 0 and 1 (index[0]=1, index[1]=1, index[3]=2)
// use iteration (array.map)? to add previous two values, and add the sum to end of array
  //array.push(sum)
// limit number of iterations with loop (while loop?) dicatated by argument provided to fibonacci(num) function
// return array value at argument location sum[num]

// And a written explanation of your solution

File: 01-week-1--starter-algorithms/02-day-3--fibonacci-series/ruby/fibonacci.rb
def fibonacci(num)
  numbers = [0, 1]

  return numbers[num] if num < 2

  (num -1).times do
    sum = numbers[0] + numbers[1]
    numbers = [numbers[1], sum]
  end

  numbers.last
  
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 0"
  puts "=>", fibonacci(0)

  puts

  puts "Expecting: 1"
  puts "=>", fibonacci(2)

  puts

  puts "Expecting: 55"
  puts "=>", fibonacci(10)

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# pseudo
# And a written explanation of your solution
